File: ruby/strings/substring_search.rb
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md

cite = "Stat rosa pristina nomine, nomina nuda tenemus."
puts("cite = #{cite}\n\n")

rosa_position = cite.index("rosa")
puts("\"rosa\" position: #{rosa_position} (#{rosa_position.class})")

odonata_position = cite.index("odonata")
puts("\"odonata\" position: #{odonata_position} (#{odonata_position.class})")

File: perl/strings/substring_search.pl
#
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
#

use strict;
use warnings;

my $cite = "Stat rosa pristina nomine, nomina nuda tenemus.";
print("cite = $cite\n\n");

my $rosa_position = index($cite, "rosa");
print("\"rosa\" position: $rosa_position\n");

my $odonata_position = index($cite, "odonata");
print("\"odonata\" position: $odonata_position\n");
